### ISSUE TITLE
Issue with `Databricks` / API Version `2023-02-01`

### ISSUE THREAD

**Issue by tombuildsstuff** [hashicorp/go-azure-sdk#436]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Service Used

`Databricks`

### API Versions Used

`2023-02-01`

### Description

Test from AzureRM:

```
------- Stdout: -------
=== RUN   TestAccDatabricksVirtualNetworkPeering_basic
=== PAUSE TestAccDatabricksVirtualNetworkPeering_basic
=== CONT  TestAccDatabricksVirtualNetworkPeering_basic
testing_new.go:86: Error running post-test destroy, there may be dangling resources: exit status 1
Error: deleting Databricks Virtual Network Peering (Subscription: "*******"
Resource Group Name: "acctestRG-databricks-230427010328653921"
Workspace Name: "acctest-ws-230427010328653921"
Virtual Network Peering Name: "acctest-230427010328653921"): performing Delete: Delete "https://management.azure.com/subscriptions/*******/resourceGroups/acctestRG-databricks-230427010328653921/providers/Microsoft.Databricks/workspaces/acctest-ws-230427010328653921/virtualNetworkPeerings/acctest-230427010328653921?api-version=2023-02-01": json: cannot unmarshal string into Go value of type odata.odata
--- FAIL: TestAccDatabricksVirtualNetworkPeering_basic (445.23s)
FAIL
```

**Comment by WodansSon** [comment]
I can get the `request` but not the `response` as that appears to be swallowed by the SDK:

```bash
DELETE /subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01 HTTP/1.1
Host: management.azure.com
User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 vnetpeering/2023-02-01) HashiCorp Terraform/1.3.9 (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
Content-Type: application/json; charset=utf-8
X-Ms-Correlation-Request-Id: 4cce649b-87d5-8544-49be-0b753715a0b1
Accept-Encoding: gzip


2023/07/28 01:00:28 [DEBUG] DELETE https://management.azure.com/subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01
2023/07/28 01:00:31 RoundTripper returned a response & error; ignoring response
2023/07/28 01:00:31 [DEBUG] Unlocking "azurerm_databricks_virtual_network_peering"
2023/07/28 01:00:31 [DEBUG] Unlocked "azurerm_databricks_virtual_network_peering"
2023-07-28T01:00:31.122-0600 [ERROR] sdk.proto: Response contains error diagnostic: diagnostic_severity=ERROR tf_proto_version=5.3 tf_rpc=ApplyResourceChange tf_req_id=b3506386-9ad9-e877-88b5-5ed8896f852f tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_databricks_virtual_network_peering
  diagnostic_summary=
  | deleting Databricks Virtual Network Peering (Subscription: "{subscription}"
  | Resource Group Name: "acctestRG-databricks-230728005600635750"
  | Workspace Name: "acctest-ws-230728005600635750"
  | Virtual Network Peering Name: "acctest-230728005600635750"): performing Delete: Delete "https://management.azure.com/subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01": json: cannot unmarshal string into Go value of type odata.odata
   diagnostic_detail=""
2023-07-28T01:00:31.141-0600 [WARN]  sdk.helper_resource: Error running Terraform CLI command:
  error=
  | exit status 1
  |
  | Error: deleting Databricks Virtual Network Peering (Subscription: "{subscription}"
  | Resource Group Name: "acctestRG-databricks-230728005600635750"
  | Workspace Name: "acctest-ws-230728005600635750"
  | Virtual Network Peering Name: "acctest-230728005600635750"): performing Delete: Delete "https://management.azure.com/subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01": json: cannot unmarshal string into Go value of type odata.odata
  |
   test_working_directory=/tmp/plugintest696332210 test_name=TestAccDatabricksVirtualNetworkPeering_basic test_terraform_path={directory}/terraform test_step_number=2
2023-07-28T01:00:31.141-0600 [DEBUG] sdk.helper_resource: Stopping providers: test_terraform_path={directory}/terraform test_step_number=2 test_working_directory=/tmp/plugintest696332210 test_name=TestAccDatabricksVirtualNetworkPeering_basic
2023-07-28T01:00:31.141-0600 [ERROR] sdk.helper_resource: Error running post-test destroy, there may be dangling resources: test_step_number=2 test_working_directory=/tmp/plugintest696332210
  error=
  | exit status 1
  |
  | Error: deleting Databricks Virtual Network Peering (Subscription: "{subscription}"
  | Resource Group Name: "acctestRG-databricks-230728005600635750"
  | Workspace Name: "acctest-ws-230728005600635750"
  | Virtual Network Peering Name: "acctest-230728005600635750"): performing Delete: Delete "https://management.azure.com/subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01": json: cannot unmarshal string into Go value of type odata.odata
  |
   test_name=TestAccDatabricksVirtualNetworkPeering_basic test_terraform_path={directory}/terraform
    testing_new.go:86: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: deleting Databricks Virtual Network Peering (Subscription: "{subscription}"
        Resource Group Name: "acctestRG-databricks-230728005600635750"
        Workspace Name: "acctest-ws-230728005600635750"
        Virtual Network Peering Name: "acctest-230728005600635750"): performing Delete: Delete "https://management.azure.com/subscriptions/{subscription}/resourceGroups/acctestRG-databricks-230728005600635750/providers/Microsoft.Databricks/workspaces/acctest-ws-230728005600635750/virtualNetworkPeerings/acctest-230728005600635750?api-version=2023-02-01": json: cannot unmarshal string into Go value of type odata.odata
```

**Comment by tombuildsstuff** [comment]
When running that via a proxy:

```
 $ https_proxy=http://localhost:8888 TF_ACC=1 go test -v ./internal/services/databricks/... -run=TestAccDatabricksVirtualNetworkPeering_basic -timeout=60m
```

Request:

```
:authority: management.azure.com
:method: DELETE
:path: /subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-databricks-230728091939815098/providers/Microsoft.Databricks/workspaces/acctest-ws-230728091939815098/virtualNetworkPeerings/acctest-230728091939815098?api-version=2023-02-01
:scheme: https
authorization: Bearer XXXXX
x-ms-correlation-request-id: 7e7dbe5b-0f10-55a4-1966-f327829999d3
content-type: application/json; charset=utf-8
user-agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 vnetpeering/2023-02-01) HashiCorp Terraform/1.6.0-dev (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 terraform-provider-azurerm/dev pid-222c6c49-1b0a-5959-a213-6608f9eb8820
accept-encoding: gzip
```

Response:

```
:status: 202
cache-control: no-cache
pragma: no-cache
content-length: 3
content-type: application/json
expires: -1
location: https://management.azure.com/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/providers/Microsoft.Network/locations/westeurope/operationResults/513e3906-fc59-4513-be9d-1c70d95a5238?api-version=2018-08-01
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
strict-transport-security: max-age=31536000; includeSubDomains
x-ms-request-id: 
x-ms-ratelimit-remaining-subscription-deletes: 14999
x-ms-correlation-request-id: 7e7dbe5b-0f10-55a4-1966-f327829999d3
x-ms-routing-request-id: GERMANYNORTH:20230728T072341Z:d5dde628-c4c4-4eec-846b-d8aea018d443
date: Fri, 28 Jul 2023 07:23:41 GMT

""
```

Guessing the issue here is going to be the empty literal string being returned for the content-type `application/json`.

**Comment by tombuildsstuff** [comment]
Fixed via #585
